samples/custom_dispatch/vulkan/shaders/example_patterns.mlir

```diff
@@ -0,0 +1,97 @@
+// RUN: iree-opt %s
+
+// The required configuration for the custom dispatch. This tells the compiler
+// the requisite target information needed to support the associated custom
+// shader.
+#spirv_target = #hal.executable.target<"vulkan", "vulkan-spirv-fb", {
+  spirv.target_env = #spirv.target_env<
+    #spirv.vce<v1.3, [Shader, GroupNonUniform], [SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers]>,
+    #spirv.resource_limits<max_compute_workgroup_size = [128, 128, 64], subgroup_size = 64>
+  >
+}>
+
+#layout = #hal.pipeline.layout<push_constants = 1, sets = [
+  <0, bindings = [
+      <0, storage_buffer, ReadOnly>,
+      <1, storage_buffer, ReadOnly>,
+      <2, storage_buffer>
+  ]>
+]>
+#bindings = [
+  #hal.interface.binding<0, 0>,
+  #hal.interface.binding<0, 1>,
+  #hal.interface.binding<0, 2>
+]
+#objects = #hal.executable.objects<{
+  #spirv_target = [
+    #hal.executable.object<{
+      path = "samples/custom_dispatch/vulkan/shaders/simple_mul.spv"
+    }>
+  ]
+}>
+
+#attrdict = {
+  export = "main",
+  layout = #layout,
+  bindings = #bindings,
+  objects = #objects
+}
+
+module {
+  pdl.pattern : benefit(1) {
```

Comment:
@stellaraccident with https://github.com/nod-ai/SHARK-Turbine/pull/108, I wanted to ask whether this still looks like a good path to go down. Essentially the idea was to allow users to specify patterns to match and replace a (flat) DAG with some custom dispatch call by specifying PDL patterns, but I'm thinking there are probably more structured ways to do it.